### ISSUE TITLE
vscode open file and directory doesnt work

### ISSUE THREAD

**Issue by colombojrj** [hyprwm/Hyprland#7722]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

No

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
hyprctl systeminfo -c
Hyprland, built from branch  at commit 9e781040d9067c2711ec2e9f5b47b76ef70762b3  (props: bump version to 0.41.1).
Date: Thu Jun 13 09:54:06 2024
Tag: v0.41.1, commits: 4818

flags: (if any)


System Information:
System name: Linux
Node name: Hohenheim
Release: 6.6.47-gentoo-dist
Version: #1 SMP PREEMPT_DYNAMIC Mon Aug 19 12:01:20 -00 2024


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation UHD Graphics 620 [8086:5917] (rev 07) (prog-if 00 [VGA controller])


os-release: NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo Linux"
ANSI_COLOR="1;32"
HOME_URL="https://www.gentoo.org/"
SUPPORT_URL="https://www.gentoo.org/support/"
BUG_REPORT_URL="https://bugs.gentoo.org/"
VERSION_ID="2.15"


plugins:

======Config-Start======


Config File: /home/junior/.config/hypr/hyprland.conf: Read Succeeded

# #######################################################################################
# AUTOGENERATED HYPR CONFIG.
# PLEASE USE THE CONFIG PROVIDED IN THE GIT REPO /examples/hypr.conf AND EDIT IT,
# OR EDIT THIS ONE ACCORDING TO THE WIKI INSTRUCTIONS.
# #######################################################################################

#
# Please note not all available settings / options are set here.
# For a full list, see the wiki
#

autogenerated = 0 # remove this line to remove the warning

# See https://wiki.hyprland.org/Configuring/Monitors/
#monitor=,preferred,auto,auto
monitor=eDP-1,1920x1080@60.03100,0x0,1               # ativa o monitor do notebook
#monitor=eDP-1,disable                               # desativa o monitor do notebook
#monitor=HDMI-A-2,1920x1080@60.03100,0x0,1           # ativa o monitor do escritorio
#monitor=HDMI-A-2, preferred, auto, 1, mirror, eDP-1 # faz mirror no monitor plugado no hdmi
#monitor=DP-1, preferred, auto, 1, mirror, eDP-1     # faz mirror no monitor plugado no vga

# See https://wiki.hyprland.org/Configuring/Keywords/ for more

# Execute your favorite apps at launch
# exec-once = waybar & hyprpaper & firefox
exec-once = hyprpaper

# Source a file (multi-file configs)
# source = ~/.config/hypr/myColors.conf

# Set programs that you use
$terminal = alacritty
$fileManager = dolphin
$menu = wofi --show drun

# Some default env vars.
env = XCURSOR_SIZE,24
env = QT_QPA_PLATFORMTHEME,qt5ct # change to qt6ct if you have that
env = QT_QPA_PLATFORM,wayland

# From https://wiki.gentoo.org/wiki/Hyprland
# xdg-desktop-portal-hyprland needs to be started after Hyprland starts
env = XDG_CURRENT_DESKTOP, Hyprland
env = XDG_SESSION_TYPE, wayland
env = XDG_SESSION_DESKTOP, Hyprland
exec-once = dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once = ~/.config/hypr/xdg-portal-hyprland

# My applications
exec-once = waybar
exec-once = nm-applet --indicator
exec-once = swaync
exec-once = insync start
exec-once = blueman-applet
exec-once = hypridle

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = br
    kb_variant = thinkpad
    kb_model =
    kb_options =
    kb_rules =

    follow_mouse = 1

    touchpad {
        natural_scroll = yes
    }

    sensitivity = 0 # -1.0 to 1.0, 0 means no modification.
}

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    gaps_in = 5
    gaps_out = 20
    border_size = 2
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)

    layout = dwindle

    # Please see https://wiki.hyprland.org/Configuring/Tearing/ before you turn this on
    allow_tearing = false
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    rounding = 10

    blur {
        enabled = true
        size = 3
        passes = 1
    }

    drop_shadow = yes
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}

animations {
    enabled = yes

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 7, myBezier
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = yes # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    new_is_master = true
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = off
}

misc {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    force_default_wallpaper = -1 # Set to 0 or 1 to disable the anime mascot wallpapers
}

# Example per-device config
# See https://wiki.hyprland.org/Configuring/Keywords/#per-device-input-configs for more
device {
    name = epic-mouse-v1
    sensitivity = -0.5
}

#xwayland {
#    force_zero_scaling = true
#}

# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
windowrulev2 = suppressevent maximize, class:.* # You'll probably like this.

# vscode rules
windowrulev2 = float,class:^(electron)$
windowrulev2=size 960 540,class:^(electron)$
windowrulev2=center,class:^(electron)$

# Flameshot rules
## important
#windowrulev2=nofullscreenrequest,class:flameshot
#windowrulev2=float,class:flameshot
#windowrulev2=monitor 0,class:flameshot
#windowrulev2=move 0 0,class:flameshot
## visual
#windowrulev2=noanim,class:flameshot
#windowrulev2=noborder,class:flameshot
#windowrulev2=rounding 0,class:flameshot

# See https://wiki.hyprland.org/Configuring/Keywords/ for more
$mainMod = SUPER

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
bind = $mainMod, Q, exec, $terminal
bind = $mainMod, C, killactive,
bind = $mainMod, M, exit,
bind = $mainMod, E, exec, $fileManager
bind = $mainMod, V, togglefloating,
bind = $mainMod, R, exec, $menu
bind = $mainMod, P, pseudo, # dwindle
bind = $mainMod, J, togglesplit, # dwindle

# Move focus with mainMod + arrow keys
bind = $mainMod, left, movefocus, l
bind = $mainMod, right, movefocus, r
bind = $mainMod, up, movefocus, u
bind = $mainMod, down, movefocus, d

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10

# Example special workspace (scratchpad)
bind = $mainMod, S, togglespecialworkspace, magic
bind = $mainMod SHIFT, S, movetoworkspace, special:magic

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow

# My shotcuts
bind = , XF86MonBrightnessDown, exec, brightnessctl set 5%-
bind = , XF86MonBrightnessUp, exec, brightnessctl set 5%+
bind = , XF86AudioRaiseVolume, exec, pactl set-sink-volume @DEFAULT_SINK@ +5%
bind = , XF86AudioLowerVolume, exec, pactl set-sink-volume @DEFAULT_SINK@ -5%
bind = , XF86AudioMute, exec, pactl set-sink-mute @DEFAULT_SINK@ toggle
bind = , XF86AudioMicMute, exec, pactl set-source-mute @DEFAULT_SOURCE@ toggle
#bind = , Print, exec, grim -g "$(slurp)"
bind = , Print, exec, /home/junior/Downloads/flameshot/build/src/flameshot gui



======Config-End========
```

</details>

### Description

Open vscode, click _Open File ..._ or _Open Folder ..._ and nothing happens. Clicking _New File ..._ works normally.

<details> <summary>vscode --verbose when I click _Open FIle_</summary>

```sh

[main 2024-09-09T16:29:14.093Z] [DialogMainService]: request to acquire file dialog lock {
  title: 'Open File',
  buttonLabel: undefined,
  filters: undefined,
  defaultPath: '/home/junior',
  properties: [ 'multiSelections', 'openFile', 'createDirectory' ]
}
[main 2024-09-09T16:29:14.094Z] [DialogMainService]: new file dialog lock created {
  title: 'Open File',
  buttonLabel: undefined,
  filters: undefined,
  defaultPath: '/home/junior',
  properties: [ 'multiSelections', 'openFile', 'createDirectory' ]
}
[main 2024-09-09T16:29:14.112Z] [DialogMainService]: file dialog lock disposed {
  title: 'Open File',
  buttonLabel: undefined,
  filters: undefined,
  defaultPath: '/home/junior',
  properties: [ 'multiSelections', 'openFile', 'createDirectory' ]
}
[main 2024-09-09T16:29:16.094Z] resolveShellEnv(): skipped (VSCODE_CLI is set)

======Config-End========
```

</details>

### How to reproduce

Basically, open vscode, click _Open File ..._ or _Open Folder ..._ and nothing happens. This also happen with vscodium.

**Comment by arash28134** [comment]
Do you have [xdg-desktop-portal-hyprland](https://github.com/hyprwm/xdg-desktop-portal-hyprland) installed?

**Comment by vaxerski** [comment]
xdph does not provide a file picker. Use something like xdg-desktop-portal-gtk alongside

**Comment by colombojrj** [comment]
Hi. Installing xdg-desktop-portal-gtk worked perfectly. Thanks very much!
